contracts/Token.sol

```diff
@@ -0,0 +1,27 @@
+pragma solidity ^0.5.0;
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20Detailed.sol";
+
+contract Token is ERC20, ERC20Detailed {
+  uint40 startTime;
+  uint40 endTime;
+  constructor(uint40 _startTime, uint40 _endTime, uint256 initialSupply) ERC20Detailed("Gold", "GLD", 18) public {
+    _mint(msg.sender, initialSupply);
+    startTime = _startTime;
+    endTime = _endTime;
+  }
+
+   event Transfer(
+    address indexed _from,
+    address indexed _to,
+    uint256 _value
+  );
+
+  function _transfer(address recipient, uint256 amount, uint256 currentTime) public returns (bool) {
```

Comment:
Best practice is to use an underscore in front of a function that is `private` or `internal`. `public` functions don't need the underscore.

Reply:
A better way would be to remove the `currentTime` argument and use `now` in the require statement. In your tests, you can start with a startTime that is a bit in the past so the test will pass. You can remove the reverts for now, I can show you how you would test for reverts with time another day.